sdk/sdktypes/trigger.go

```diff
@@ -88,11 +88,24 @@ func (p Trigger) ToValues() map[string]Value {
 	return map[string]Value{
 		"name": NewStringValue(p.read().Name),
 		"data": kittehs.Must1(NewStructValue(
-			NewStringValue("event_data"),
+			NewStringValue("trigger_data"),
 			kittehs.TransformMapValues(
 				p.read().Data,
 				forceFromProto[Value],
 			),
 		)),
 	}
 }
+
+func (p Trigger) WithUpdatedData(key string, val Value) Trigger {
```

Comment:
WithData is more idiomatic

Reply:
This is a deliberate naming decision. 
Otherwise it will be unaligned with our naming convention for getters and setters of sdktypes. `WithData` should update all data and receive a map, see `events.WithData()`. The same logic is for getters - `Data()` returns a map and not a single key/value. Therefore, since I wanted to update only a single key I called this method `withUpdatedData(key, value)`.

Reply:
WithAdditionalData